tools/fetch-sponsors.js

```diff
@@ -107,7 +107,7 @@ async function fetchOpenCollectiveData() {
             }
           }
   
-          cancelledOrders: orders(filter: INCOMING, onlySubscriptions: true, dateFrom:"${startOfCurrentMonth}") {
+          cancelledOrders: orders(status: CANCELLED, filter: INCOMING, onlySubscriptions: true, dateFrom:"${startOfCurrentMonth}") {
```

Comment:
I tested this with hardcoded `2023-05-01T00:00:00Z`:

<details>

<summary>result</summary>

```json
{
    "totalCount": 3,
    "nodes": [
        {
            "fromAccount": {
                "name": "iBoysoft",
                "website": "https://iboysoft.com/",
                "imageUrl": "https://images.opencollective.com/iboysoft-software/7f9d60e/avatar.png"
            },
            "amount": {
                "value": 200
            },
            "tier": {
                "slug": "bronze-sponsor"
            },
            "frequency": "MONTHLY",
            "totalDonations": {
                "value": 0
            }
        },
        {
            "fromAccount": {
                "name": "BairesDev",
                "website": "https://www.bairesdev.com/sponsoring-open-source-projects/",
                "imageUrl": "https://images.opencollective.com/bairesdev/48bb773/logo.png"
            },
            "amount": {
                "value": 200
            },
            "tier": {
                "slug": "bronze-sponsor"
            },
            "frequency": "MONTHLY",
            "totalDonations": {
                "value": 200
            }
        },
        {
            "fromAccount": {
                "name": "King Billy Slots",
                "website": null,
                "imageUrl": "https://images.opencollective.com/king-billy-slots1/c30c2aa/avatar.png"
            },
            "amount": {
                "value": 200
            },
            "tier": {
                "slug": "bronze-sponsor"
            },
            "frequency": "MONTHLY",
            "totalDonations": {
                "value": 0
            }
        }
    ]
}

```

</details>

Two problems:

1. It still includes rejected (refunded) ones like the slots site, perhaps we could filter them out by `totalDonations === 0`.
2. I was expecting to see PayDay Say (https://opencollective.com/eslint/transactions?searchTerm=PayDay) but it didn't appear. It appears with `2023-01-01T00:00:00Z`. Looks like `dateFrom` means the sponsorship start date, but what we want is the sponsorship cancellation / last payment date?

Reply:
I'll contact OC about the potential bug with returning rejected sponsorships.

`dateFrom` is the date after which the activity occurred. Because PayPay Say cancelled in May, I wouldn't expect it to show up when date from starts on July 1. So we'd only get sponsorships cancelled on or after July 1.

Reply:
> Because PayPay Say cancelled in May, I wouldn't expect it to show up when date from starts on July 1. So we'd only get sponsorships cancelled on or after July 1.

Yes, it shouldn't show up after May, but it seems that it wouldn't show up in May (which is what we want if I'm not misunderstanding the intent).

Reply:
KingBilly Slots wasn't rejected. It's a bit weird as the record shows it was refunded before the donation was made, so I think there may be an issue on the OC side with these records. I'm not seeing any sponsors that were rejected even if I go back to the beginning of the year.

I don't think we can make any determination about whether PayDay Say is a bug or not because we don't know when they actually cancelled their sponsorship. We do know this is working for iBoySoft, so I'd say let's go with this version of the script and see what happens.

Reply:
Okay, I still don't think this works as we'd like it to, but let's merge this and see what happens next time when the new `cancelledOrders` query returns something (currently, it returns empty list).